controllers/etcd_copy_backups_controller.go

```diff
@@ -0,0 +1,409 @@
+	if _, ok := etcdCopyBackupsTask.Annotations[v1beta1constants.GardenerOperation]; ok {
+		objectCopy := etcdCopyBackupsTask.DeepCopy()
+		annotations := objectCopy.GetAnnotations()
+		delete(annotations, v1beta1constants.GardenerOperation)
+		objectCopy.SetAnnotations(annotations)
+		if err := r.Patch(ctx, objectCopy, client.MergeFrom(etcdCopyBackupsTask)); client.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) getCopyJob(etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask, values map[string]interface{}, logger logr.Logger) (*batchv1.Job, error) {
+	decoded := &batchv1.Job{}
+	copyJobPath := filepath.Join("etcd", "templates", "etcd-copy-backups-job.yaml")
+	chartPath := getChartPath()
+
+	renderedChart, err := r.chartApplier.Render(chartPath, etcdCopyBackupsTask.Name, etcdCopyBackupsTask.Namespace, values)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := decodeObject(renderedChart, copyJobPath, &decoded); err != nil {
+		return nil, err
+	}
+
+	return decoded, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) runCopyJob(ctx context.Context, etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask, values map[string]interface{}, logger logr.Logger) (*batchv1.Job, error) {
+	job, err := r.getCopyJob(etcdCopyBackupsTask, values, logger)
+	if err != nil {
+		return nil, err
+	}
+	return job, r.Create(ctx, job)
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) getValuesMap(etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask) (map[string]interface{}, error) {
+	values := map[string]interface{}{
+		"jobName":  etcdCopyBackupsTask.Name + workerSuffix,
+		"taskName": etcdCopyBackupsTask.Name,
+		"uid":      etcdCopyBackupsTask.UID,
+	}
+
+	sourceStoreValues, err := utils.GetStoreValues(etcdCopyBackupsTask.Spec.SourceStore)
+	if err != nil {
+		return nil, err
+	}
+	targetStoreValues, err := utils.GetStoreValues(etcdCopyBackupsTask.Spec.TargetStore)
+	if err != nil {
+		return nil, err
+	}
+	values["sourceStore"] = sourceStoreValues
+	values["targetStore"] = targetStoreValues
+
+	if etcdCopyBackupsTask.Spec.MaxBackupAge != nil {
+		values["maxBackupAge"] = *etcdCopyBackupsTask.Spec.MaxBackupAge
+	}
+	if etcdCopyBackupsTask.Spec.MaxBackups != nil {
+		values["maxBackups"] = *etcdCopyBackupsTask.Spec.MaxBackups
+	}
+
+	if etcdCopyBackupsTask.Spec.Image != nil {
+		values["image"] = etcdCopyBackupsTask.Spec.Image
+	} else {
+		images, err := imagevector.FindImages(r.imageVector, []string{common.BackupRestore})
+		if err != nil {
+			return map[string]interface{}{}, err
+		}
+		val, ok := images[common.BackupRestore]
+		if !ok {
+			return nil, errors.New("Unable to find etcdbrctl image")
+		}
+		values["image"] = val.String()
+	}
+
+	return values, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) addFinalizerToObject(ctx context.Context, object client.Object, logger logr.Logger) error {
```

Comment:
This can be a reusable and tested util function if you remove the pointer receiver and pass the client.